test/apis/testing/v1alpha1/build_types.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2018 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/knative/pkg/apis/duck"
+	duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Build is a simple test build resource.
+type Build struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec holds the desired state of the Build (from the client).
+	// +optional
+	Spec BuildSpec `json:"spec,omitempty"`
+
+	// Status communicates the observed state of the Build (from the controller).
+	// +optional
+	Status BuildStatus `json:"status,omitempty"`
+}
+
+// Check that ConfigurationStatus may have its conditions managed.
+var _ duckv1alpha1.ConditionsAccessor = (*BuildStatus)(nil)
+
+// Check that Build implements the Conditions duck type.
+var _ = duck.VerifyType(&Build{}, &duckv1alpha1.Conditions{})
+
+// BuildSpec holds the desired state of the Build (from the client).
+type BuildSpec struct {
+	// TODO(mattmoor): ???
```

Comment:
I'd like something like
```
	Fail bool `json:"fail,omitempty"`
```
here, and make the controller fail the build if `Fail == true`

Reply:
On the second thought, specifying Conditions to be added during upcoming reconciliation cycles, would probably be a better idea

Reply:
I added a way to specify the failure reason/message (success if omitted)